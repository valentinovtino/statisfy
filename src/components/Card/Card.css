 .card {
    width: 260px;
    height: 333px;
    border: 1px solid red;
    padding: 10px;
    text-align: center;
    margin: 20px;
    box-shadow: 5px 7px 5px black;
    opacity: .8; 
    display: flex;
    flex-direction: column;
}

.food-img {
    height: 100%;
    width: 100%;
}

.food-title {
    font-size: small;
}

.top {
    border: 1px solid blue;
    display: flex;
    flex-direction: row;
}

.vote-btns {
    display: flex;
    flex-direction: column;
}

.bottom {
    border: 1px solid green;
}