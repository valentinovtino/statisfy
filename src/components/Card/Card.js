import React, {Component} from 'react';
import './Card.css';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { connect } from "react-redux";
import { addToFavorites, removeFavorite } from '../../actions/actions';

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      test: ''
    };
  }


  render() {
    const { food, fave, location } = this.props;
    
    const button = location.pathname === '/fave' ? <button className='like-btn'onClick={() => this.props.removeFavorite(this.props.food)}>REMOVE</button> :
    <button className='like-btn'onClick={() => this.props.addToFavorites(this.props.food)}>FAVE</button>;

    return (
      <div className="card" id={food.recipe_id}>
        <div className="top">
          <div className='food-container'>
            <img className="food-img" src={food.image_url}/>
          </div>
          <div className='vote-btns'>
            <img className='vote' src='https://cdn3.iconfinder.com/data/icons/pixo-icons-2/56/1-29-512.png'/>
            {button}
          </div>
        </div>

        <div className='bottom'>
          <h1 className="food-title">{food.title}</h1>
          <a className='link' href={food.source_url}>Recipe</a>
        </div>
      </div> 
      
    );
  }
}

Card.propTypes = {
  addToFavorites: PropTypes.func,
  removeFavorite: PropTypes.func,
  food: PropTypes.Object,
  fave: PropTypes.func,
  allState: PropTypes.Object
};

export const mapStateToProps = (state) => ({
  category: state.category
});

export const mapDispatchToProps = (dispatch) => ({
  addToFavorites: (food) => dispatch(addToFavorites(food)),
  removeFavorite: (food, title) => dispatch(removeFavorite(food, title))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Card));

