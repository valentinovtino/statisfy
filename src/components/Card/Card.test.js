import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';
import ReactDOM from 'react-dom';

describe('Card', () => {
  let wrapper;
  let mockFood = {};
  let mockFunction = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Card food={mockFood} />);
  });
  it('should render a card', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFavorite when Add to favorites button is clicked', () => {
    wrapper = mount(<Card addToFavorites={mockFunction} food={mockFood} />);

    const spy = spyOn(wrapper.instance(), 'handleFavorite');

    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it.skip('should call handleRemoveFavorite when Remove favorites button is clicked', () => {
    wrapper = mount(<Card removeFavorite={mockFunction} food={mockFood} />);

    const spy = spyOn(wrapper.instance(), 'handleRemoveFavorite');

    wrapper.instance().forceUpdate();
    wrapper.find('.remove-btn').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

 
});