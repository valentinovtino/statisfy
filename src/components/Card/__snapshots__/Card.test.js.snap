// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card 1`] = `undefined`;
